### PR TITLE
T353880: Create page-change eventstream client

### PR DESCRIPTION
**Pull request by MobyDickIsAfrican**
Ticket: https://phabricator.wikimedia.org/T353880

Added page-change stream client. Unit tests and example work.

### CODE REVIEW

File: page_change.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package eventstream
+
+import (
+	"encoding/json"
+	"time"
+)
+
+// RevisionCreate event scheme struct
```
Comment (prabhat393) [inline page_change.go]: Pls update the comment.

File: page_change.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+			PageID    int    `json:"page_id"`
+			PageTitle string `json:"page_title"`
+		} `json:"page"`
+		Revision struct {
+			RevID int       `json:"rev_id"`
+			RevDt time.Time `json:"rev_dt"`
+		} `json:"revision"`
+		WikiID string `json:"wiki_id"`
+	}
+}
```
Comment (prabhat393) [inline page_change.go]: There is a lot of schema changes between the older streams (such as [revision-create)](https://schema.wikimedia.org/repositories//primary/jsonschema/mediawiki/revision/create/latest.yaml) vrs the new [page-change](https://schema.wikimedia.org/repositories//primary/jsonschema/mediawiki/page/change/latest.yaml) stream.

In order to accommodate both, I suggest the following changes:

1. Include `page_change_kind` as a PageChange.Data field. 

2. As you can see in both the schemas, revision fields are restructured quite a bit.
In the old schema, revision fields (such as rev_id) is in the root. In the new schema, the revision fields are structured under a revision struct.
To work with both, take out `RevID` from the `baseData`. Add `RevID` to all the old structs (such as RevisionCreate).Data as a field.
For the new struct PageChane.Data.Revision add all the following fields:
     - rev_id
     - rev_dt
     - comment
     - content_slots.content_format
     - content_slots.content_model
     - is_comment_visible
     - is_content_visible
     - is_editor_visible
     - is_minor_edit
     - rev_parent_id
     - rev_sha1
     - rev_size
     
 3. Page fields are restructured as well. Remove the following from `baseData`:
      
```
        PageID         int       `json:"page_id"`
	PageTitle      string    `json:"page_title"`
	PageNamespace  int       `json:"page_namespace"`
	PageIsRedirect bool      `json:"page_is_redirect"`
```

	
Add these to all the older structs (such as RevisionCreate).Data as a field there.
	
For the new struct (PageChange), also include the missing namespace and redirect.

4. `Database` is renamed in the new schema. Remove the following from `baseData`:
       
       `Database       string    `json:"database"``
       
  Add it to all the older structs (such as RevisionCreate).Data as a field there.
       
  For the new struct (PageChange), add the following under .Data
       
      ` Database       string    `json:"wiki_id"``

File: revision_score.go [hunk 3]
```diff
@@ -10,9 +10,14 @@ type RevisionScore struct {
 	baseSchema
```
Comment (prabhat393) [inline revision_score.go]: `revision-score` stream is no longer supported.

https://stream.wikimedia.org/?doc#/streams

Let's remove `revision_score.go`, it's unit test, from client.go, options.go, builder.go and references from their unittest.

File: utils.go [hunk 4]
```diff
@@ -48,6 +48,5 @@ func readStub(name string) ([][]byte, error) {
 		msg += "data: " + string(data) + "\n"
 		msgs = append(msgs, []byte(msg))
 	}
-
```
Comment (prabhat393) [inline utils.go]: Nit-pick: keep the empty line before `return`

File: revision_score.go (hunk 3 above)
Comment (WikiCreatorOfWorlds) [reply revision_score.go]: Done.

File: utils.go (hunk 4 above)
Comment (WikiCreatorOfWorlds) [reply utils.go]: Done.

File: client.go [hunk 5]
```diff
@@ -18,6 +18,7 @@ const (
 	revisionCreateURL           = "/v2/stream/revision-create"
 	revisionScoreURL            = "/v2/stream/revision-score"
```
Comment (prabhat393) [inline client.go]: Let's remove `revisionScoreURL`

**Review by resquito — APPROVED** [review]

**Pull request merged**